packages/theme-dark-high-contrast-extension/style/variables.css

```diff
@@ -437,4 +437,18 @@ all of MD as it is not optimized for dense, information rich UIs.
   --jp-inspector-icon-color: var(--md-grey-200);
   --jp-switch-color: var(--md-grey-400);
   --jp-switch-true-position-color: var(--md-orange-700);
+
+  /* Completer specific styles */
+
+  --jp-completer-type-background0: transparent;
+  --jp-completer-type-background1: #1f77b4;
+  --jp-completer-type-background2: #ff7f0e;
+  --jp-completer-type-background3: #2ca02c;
+  --jp-completer-type-background4: #d62728;
+  --jp-completer-type-background5: #9467bd;
+  --jp-completer-type-background6: #8c564b;
+  --jp-completer-type-background7: #e377c2;
+  --jp-completer-type-background8: #7f7f7f;
+  --jp-completer-type-background9: #bcbd22;
+  --jp-completer-type-background10: #17becf;
```

Comment:
What I would ultimately like to do is to stop putting things into `:root` when they do not need to be there. This is because performance suffers doubly:
- from browser having too many variables declared in a scope (this is a small cumulative effect)
- from them being set on global scope when a local one would do (this is a problem especially in Chrome, see https://issuetracker.google.com/issues/40676722)

Does this one work?

```suggestion
}

/* Completer specific styles */
  
.jp-Completer {
  --jp-completer-type-background0: transparent;
  --jp-completer-type-background1: #1f77b4;
  --jp-completer-type-background2: #ff7f0e;
  --jp-completer-type-background3: #2ca02c;
  --jp-completer-type-background4: #d62728;
  --jp-completer-type-background5: #9467bd;
  --jp-completer-type-background6: #8c564b;
  --jp-completer-type-background7: #e377c2;
  --jp-completer-type-background8: #7f7f7f;
  --jp-completer-type-background9: #bcbd22;
  --jp-completer-type-background10: #17becf;
```

I personally do not see that much of an advantage in having to set a variable vs having to write a three-line CSS, My reluctance is because these variables are used only once in the entire CSS codebase but have a non-obvious cost which only becomes apparent when you work with large notebooks and get into the weeds of browser performance profiling. I spent a lot of time on it with https://github.com/jupyterlab/ui-profiler and more low-level tools and saw the `:root` group of variables show up quite often as a potential culprit of UI freezes.

Reply:
Very interesting, thank you very much for sharing these resources! Proposed changes implemented.

To be honest, in this case, I was prioritizing consistency given how the other variables are organized. Having everything scoped out in `:root` is quite useful for collaborating with multiple people from different backgrounds because, from my perspective, it makes the CSS look like just a configuration file. Anyway, I agree with you.

As a follow-up to this PR, I would like to suggest opening an issue to discuss the reorganization of the variable files for the various themes, as well as adding a note to the [documentation](https://jupyterlab.readthedocs.io/en/stable/developer/css.html#css-variables) about this organization (there is actually already a point about this, but I think it can be extended): _critical_ variables must be scoped in `:root` , the rest in their respective parent classes (with one example). What do you think?

Reply:
Good call, let's continue the discussion in https://github.com/jupyterlab/jupyterlab/issues/16660